src/storage.ts

```diff
@@ -4,30 +4,61 @@ interface ClientStorageOptions {
   daysUntilExpire: number;
 }
 
-export const getAllKeys = () => Object.keys(Cookies.getAll() || {});
+/**
+ * Defines a type that handles storage to/from a storage location
+ */
+export type ClientStorage = {
+  get<T extends Object>(key: string): T;
+  save(key: string, value: any, options?: ClientStorageOptions): void;
+  remove(key: string): void;
+};
+
+/**
+ * A storage protocol for marshalling data to/from cookies
+ */
+export const CookieStorage = {
+  get<T extends Object>(key: string) {
+    const value = Cookies.get(key);
+    if (typeof value === 'undefined') {
+      return;
+    }
+    return <T>JSON.parse(value);
+  },
+
+  save(key: string, value: any, options?: ClientStorageOptions): void {
+    let cookieAttributes: Cookies.CookieAttributes = {};
+    if ('https:' === window.location.protocol) {
+      cookieAttributes = {
```

Comment:
People affected by #515 will still have an issue with the `is.authenticated` cookie

We'll need to do the [2 cookie solution](https://web.dev/samesite-cookie-recipes/#handling-incompatible-clients) on that before we can say it's fixed

Reply:
They should no longer get the invalid state issue though, which is what he original issue reported.

Reply:
true, we can handle that in another PR - but we should include it in the next release

i've started a branch here https://github.com/auth0/auth0-spa-js/tree/legacy-samesite